src/jquery-json-editor.js

```diff
@@ -526,6 +547,185 @@
+                        value: "× Delete selected possible values",
+                        on: {
+                            click: function () {
+                                $possibleValuesSelect.children("option:selected").remove();
+                            }
+                        }
+                    });
+
+                    /*!
+                     * nameAlreadyExists
+                     * A function that determines whether the given name
+                     * already exists in a field under the path in which the
+                     * user tries to create a new field.
+                     *
+                     * @name nameAlreadyExists
+                     * @function
+                     * @param {String} name The name of the field to search for.
+                     * @return {Boolean} True if the name already exists, false
+                     * otherwise.
+                     */
+                    function nameAlreadyExists(name) {
+                        // Convert the array of jQuery elements to a jQuery
+                        // object with multiple elements.
+                        var $input2 = $($.map($input,
+                                    function (e) { return e.get(0); }));
+
+                        // Obtain the data at the path where the new field is
+                        // created.
+                        var data = self.getData(field.path, $input2);
+
+                        // Return true if the given name is already in the
+                        // data, otherwise return false.
+                        return Object.keys(data).indexOf(name) > -1;
+                    }
+
+                    var $addFieldButton = $("<input>", {
+                        type: "button",
+                        value: "+ Add field",
+                        on: {
+                            click: function () {
+                                $nameInput.val($nameInput.val()
+                                        .replace(/\./g, "").trim());
+                                $labelInput.val($labelInput.val().trim());
+
+                                var name = $nameInput.val();
+                                var label = $labelInput.val();
+
+                                // Validate.
+                                if (name === "+" || name.length === 0 ||
+                                        nameAlreadyExists(name)) {
+                                    alert("The name of the field should be a" +
+                                            " non-empty string without dots, " +
+                                            "different than \"+\" and not " +
+                                            "already existing under the " +
+                                            "path \"" + field.path + "\".");
+                                    return;
+                                }
+                                if (label.length === 0) {
```

Comment:
Instead do this: `label = label || name` so we won't show violent messages to the user.